src/hooks/usePoolData.ts

```diff
@@ -312,32 +310,19 @@ export default function usePoolData(
       }
 
       const metaSwapAddress = POOL.metaSwapAddresses?.[chainId]
+      const poolAddressToCheckMigrationStatus = metaSwapAddress || poolAddress
+      const migrationAddress = metaSwapAddress ? metaSwapAddress : poolAddress
       let isMigrated = false
-      if (metaSwapAddress) {
-        try {
-          const migrationMapRes = await migratorContract.migrationMap(
-            metaSwapAddress,
-          )
-          isMigrated =
-            migrationMapRes.oldPoolLPTokenAddress !==
-            "0x0000000000000000000000000000000000000000"
-              ? true
-              : false
-        } catch (err) {
-          console.error(err)
-        }
-      }
-      // Edge case for Stable Coin Pool as it has no metaSwapAddress but is still migrated.
-      if (poolName === STABLECOIN_POOL_NAME) {
+      if (
+        poolAddressToCheckMigrationStatus &&
+        migratorContract &&
+        migrationAddress
+      ) {
         try {
           const migrationMapRes = await migratorContract.migrationMap(
-            poolAddress,
+            migrationAddress,
           )
-          isMigrated =
-            migrationMapRes.oldPoolLPTokenAddress !==
-            "0x0000000000000000000000000000000000000000"
-              ? true
-              : false
+          isMigrated = migrationMapRes.oldPoolLPTokenAddress !== AddressZero
```

Comment:
is it fine to do the check against oldPoolLPTokenAddress?

Reply:
any of the fields should be fine, I just went with the shortest key for readability.